crates/proof-of-sql-parser/src/posql_time/unit.rs

```diff
@@ -15,6 +15,40 @@ pub enum PoSQLTimeUnit {
     Nanosecond,
 }
 
+impl PoSQLTimeUnit {
+    /// Convert the numerical unit of one timeunit to another for comparison purposes.
+    pub fn normalize_timeunit(
+        timestamp: i64,
+        from_unit: &PoSQLTimeUnit,
+        to_unit: &PoSQLTimeUnit,
+    ) -> i64 {
+        match (from_unit, to_unit) {
+            // Conversions from Seconds
+            (PoSQLTimeUnit::Second, PoSQLTimeUnit::Millisecond) => timestamp * 1000,
+            (PoSQLTimeUnit::Second, PoSQLTimeUnit::Microsecond) => timestamp * 1_000_000,
+            (PoSQLTimeUnit::Second, PoSQLTimeUnit::Nanosecond) => timestamp * 1_000_000_000,
+
+            // Conversions from Milliseconds
+            (PoSQLTimeUnit::Millisecond, PoSQLTimeUnit::Second) => timestamp / 1000,
```

Comment:
Hmm if we accept rounding then 10AM and 10AM plus 1ms will be identical if we normalize to s.

Reply:
Ok no rounding, got it